src/Avalonia.Base/Input/GestureRecognizers/HoldGestureRecognizer.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Timers;
+using Avalonia.Input.GestureRecognizers;
+using Avalonia.Threading;
+
+namespace Avalonia.Input
+{
+    public class HoldGestureRecognizer : StyledElement, IGestureRecognizer
+    {
+        private const int Tolerance = 30;
+        private IInputElement? _target;
+        private IGestureRecognizerActionsDispatcher? _actions;
+        private int _gestureId;
+        private IPointer? _tracking;
+        private PointerPressedEventArgs? _pointerEventArgs;
+        private Rect _trackingBounds;
+        private Timer? _holdTimer;
+        private bool _elasped;
+
+        /// <summary>
+        /// Defines the <see cref="IsHoldWithMouseEnabled"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsHoldWithMouseEnabledProperty =
+            AvaloniaProperty.Register<HoldGestureRecognizer, bool>(
+                nameof(IsHoldWithMouseEnabled));
+        
+        /// <summary>
+        /// Gets or sets whether to detect hold from the mouse
+        /// </summary>
+        public bool IsHoldWithMouseEnabled
+        {
+            get => GetValue(IsHoldWithMouseEnabledProperty);
+            set => SetValue(IsHoldWithMouseEnabledProperty, value);
+        }
+
+        public void Initialize(IInputElement target, IGestureRecognizerActionsDispatcher actions)
+        {
+            _target = target;
+            _actions = actions;
+
+            _target?.AddHandler(InputElement.PointerPressedEvent, OnPointerPressed, Interactivity.RoutingStrategies.Tunnel | Interactivity.RoutingStrategies.Bubble);
+            _target?.AddHandler(InputElement.PointerReleasedEvent, OnPointerReleased, Interactivity.RoutingStrategies.Tunnel | Interactivity.RoutingStrategies.Bubble);
+
+            _holdTimer = new Timer(300);
+            _holdTimer.AutoReset = false;
+            _holdTimer.Elapsed += HoldTimer_Elapsed;
```

Comment:
@kekekeks @grokys what do you think about using DispatcherTimer instead?
For two reasons:
- we anyway will need to jump back to the UI thread with clr Timer
- I don't really know if DispatcherTimer will have less overhead or not, please confirm